components/hab/src/main.rs

```diff
@@ -304,10 +305,33 @@ fn sub_pkg_export(ui: &mut UI, m: &ArgMatches) -> Result<()> {
 }
 
 fn sub_pkg_hash(m: &ArgMatches) -> Result<()> {
-    let source = m.value_of("SOURCE").unwrap(); // Required via clap
-
     init();
-    command::pkg::hash::start(&source)
+    match m.value_of("SOURCE") {
+        Some(source) => {
+            // hash single file
+            command::pkg::hash::start(&source)
+        },
+        None => {
+            // read files from stdin
+            let stdin = io::stdin();
+            for line in stdin.lock().lines() {
+                match line {
+                    Ok(line) => {
+                        let file = line.trim_right();
+                        if Path::new(&file).is_file() {
+                            command::pkg::hash::start_multi(&file);
```

Comment:
This feels just a bit awkward to me to have 2 functions (`start` and `start_multi`) where one differs just in output, but I see why you are doing it. I wonder if it would be better to simply call `hash::hash_file(&file)` here and then spit out our output.